node/src/components/consensus/highway_core/finality_detector/rewards.rs

```diff
@@ -23,14 +29,18 @@ pub(crate) fn compute_rewards<C: Context>(state: &State<C>, bhash: &C::Hash) ->
     let mut rewards = ValidatorMap::from(vec![0u64; panorama.len()]);
```

Comment:
You can just return here instead of checking in every iteration of the loop below. The rewards are already initialized with 0.

```rust
if !should_compute_rewards {
    return rewards;
}
```

Reply:
Applied, thanks